guide/installation-and-configuration/upgrading-to-37.markdown

```diff
@@ -73,38 +156,31 @@ If your clients get promise failures (not kept) similar to "Can't stat file '/va
+  * Identify existing modifications to the masterfiles directory.  If patches from version control are unavailable or require verification, a copy of /var/cfengine/masterfiles from a clean installation of your previous version can help identify changes which will need to be applied to a new 3.7 install.
+  * The 3.7 masterfiles can be found in a clean installation of CFEngine (hub package on Enterprise), under /var/cfengine/masterfiles.  Apply any customizations against a copy of the 3.7 masterfiles in a well-known location, e.g. `/root/3.7/masterfiles`.
+  * Use `cf-promises` to verify that the policy runs with 3.7, by running `cf-promises /root/3.7/masterfiles/promises.cf` and `cf-promises /root/3.7/masterfiles/update.cf`.
+  * Use `cf-promises` to verify that the policy runs with you previous version of CFEngine (e.g. 3.6), by running the same commands as above on a node with that CFEngine version.
+  * The merged masterfiles should now be based on the 3.7 framework, include your policies and work on both the version you are upgrading from and with 3.7.
 2. Set `trustkeysfrom` to trust all hosts, or at least the Policy Server, 
-    in the merged masterfiles policy, e.g. /root/3.6/masterfiles/update/update_policy.cf`.
-
-    CFEngine 3.6 uses LMDB for local databases, whereas older versions of CFEngine typically use TokyoCabinet or QDBM. The classic networking protocol uses the `lastseen` database to verify that the mapping between a peer's IP address and the corresponding hostkey is not changed. Since the 3.6 installation will not have any mappings in the lastseen database, hosts won't trust the IP address of the policy server without that setting.
-3. Optional, Enterprise only: Export the data from your existing Enterprise MongoDB.
-  * Download the [`cfmigrate`](http://cfengine.package-repos.s3.amazonaws.com/tools/cfmigrate) binary.
-  * This binary will export user/role settings as well as long-living file-changes data from MongoDB.
-  * No other data will be exported, as it would either way expire after
-    one week. If you need continued access to 3.5 compliance data,
-    keep the 3.5 MongoDB available.
-  * The `cfmigrate` tool uses `mongoexport`, which is available from [mongodb.org](http://www.mongodb.org/downloads) (it is included in the hub package of CFEngine 3.5).
-4. On your existing Policy Server, stop the CFEngine services.
+    in the merged masterfiles policy, e.g. /root/3.7/masterfiles/update/update_policy.cf`.
+
+    Since 3.6, CFEngine uses LMDB for local databases, whereas older versions of CFEngine typically use TokyoCabinet or QDBM. The classic networking protocol uses the `lastseen` database to verify that the mapping between a peer's IP address and the corresponding hostkey is not changed. Since the 3.7 installation will not have any mappings in the lastseen database, hosts won't trust the IP address of the policy server without that setting.
```

Comment:
I think this paragraph can be completely removed cant it? 3.6 had LMDB, so 3.7 wont have this issue of missing entries.